src/RedisTaggedCache.js

```diff
@@ -104,6 +105,39 @@ class RedisTaggedCache extends TaggedCache {
       });
   }
 
+  /**
+   * Bulk operation on tagged entries.
+   *
+   * @param {Operation} operation
+   *   The operation to perform on each individual item.
+   * @param {number} pageSize
+   *   The number of operations to execute at once. Empty to execute all.
+   *
+   * @return {Promise<[any]>}
+   *   Resolves the result of the individual operations.
+   */
+  bulk<T>(
+    operation: Operation<T>,
+    pageSize: number = Infinity
+  ): Promise<Array<T>> {
+    return this.fetchTaggedKeysByTag()
+      .then(this.fetchTaggedKeys.bind(this))
+      .then(keys =>
+        promiseMap(keys, key => operation(key, this), { concurrency: pageSize })
+      );
+  }
+
+  /**
+   * Sets all the tagged entries to expire so they can be evicted by Redis.
+   *
+   * @return {Pr}
+   */
+  invalidateWithTags(): Promise<Array<void>> {
+    const invalidate: Operation<void> = (key, tc) =>
```

Comment:
This is problematic because it won't delete the cache key from the tag Set.

Reply:
Is it problematic to the point where this should be rolled back? If expired items aren't automatically removed from the set by redis this could lead to a lot of orphaned tags.

Reply:
Yeah, I have a PR in the works. It's just that some other stuff is taking precedence. Since it's a new API I don't think this breaks anyone yet.